zerver/lib/thumbnail.py

```diff
@@ -453,9 +453,18 @@ def rewrite_thumbnailed_images(
         path_id = image_link["href"].removeprefix("/user_uploads/")
         if to_delete and path_id in to_delete:
             # This was not a valid thumbnail target, for some reason.
-            # Trim out the whole "message_inline_image" element, since
+
+            # image has title, trim out the whole "message_inline_image" element, since
             # it's not going be renderable by clients either.
-            inline_image_div.decompose()
+            if image_link.get("title"):
+                inline_image_div.decompose()
+            else:
+                # image with no title, replace "message_inline_image" with a non-image form,
+                # so the file is still accessible.
+                failed_thumbnail_rendered_content = BeautifulSoup(
+                    '<p><a href="{url}">{url}</a></p>'.format(url=image_link["href"]), "html.parser"
+                )
```

Comment:
This is not the right way to insert a new tag.  It has escaping problems and is inefficient.

See https://www.crummy.com/software/BeautifulSoup/bs4/doc/#navigablestring-and-new-tag

Though I would ideally also like some way to make sure this result agrees (in terms of message structure) with the result of if the user had originally written `[](/user_uploads/12/foo/not-an-image)` in the message.  I don't _think_ it's the right solution (and I think we don't have enough context to be able to do so), but we almost want delete the ImageAttachment row and then re-render the whole markdown contents, which would result in it not being treated as a thumbnail-able image, like we want.  And then we don't need to keep this rendering consistent with what "would have" happened if we'd never thought it was an image.